internal/state/dataplane/configuration.go

```diff
@@ -150,12 +154,7 @@ func buildWarnings(graph *graph.Graph, upstreams map[string]Upstream) Warnings {
 				continue
 			}
 
-			for _, group := range r.BackendGroups {
-
-				for _, errMsg := range group.Errors {
-					warnings.AddWarningf(r.Source, "invalid backend ref: %s", errMsg)
-				}
-
+			for _, group := range r.GetAllBackendGroups() {
```

Comment:
I got it wrong. We don't report it in the condition because upstream building and resolving is done in the `dataplane` package.

However, this should be reported as a condition.

I'd rather not further increase the scope of this PR. How about a FIXME?

Reply:
FIXME works 👍

Reply:
https://github.com/nginxinc/nginx-kubernetes-gateway/pull/455/commits/c9f5834fbb7a6b68fea82237823b12c4c89f4c89